src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -634,11 +671,32 @@ private void RenderInExistingBatch(RenderQueueEntry renderQueueEntry)
                 var disposeComponentId = _batchBuilder.ComponentDisposalQueue.Dequeue();
                 var disposeComponentState = GetRequiredComponentState(disposeComponentId);
                 Log.DisposingComponent(_logger, disposeComponentState);
-                if (!disposeComponentState.TryDisposeInBatch(_batchBuilder, out var exception))
+                if (disposeComponentState.RequiresAsyncDisposal())
```

Comment:
Should this `if` condition be inverted? Looks like the "true" branch below deals with the synchronous disposal case.

Reply:
cleaned up